pkg/resourcemanager/controller/garbagecollector/reconciler.go

```diff
@@ -70,7 +73,37 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
 				continue
 			}
 
-			objectsToGarbageCollect[objectId{resource.kind, obj.Namespace, obj.Name}] = struct{}{}
+			if suppressString := obj.Annotations[managedresourcesutils.AnnotationKeySuppressGarbageCollectionUntil]; suppressString != "" {
+				suppressUntilTime, err := time.Parse(time.RFC3339, suppressString)
+				if err != nil {
+					log.Error(
+						err,
+						fmt.Sprintf( //nolint:logcheck // reason: Linter does not realise that message string is a constant
+							"Garbage collector: collective object has invalid '%s' annotation",
+							managedresourcesutils.AnnotationKeySuppressGarbageCollectionUntil),
```

Comment:
```suggestion
						"Invalid timestamp for garbage collection suppression annotation",
```

Nit, should be good enough.